src/cpp/src/model_runner.hpp

```diff
@@ -153,6 +173,7 @@ class ModelRunner {
                 subsequence_begins_data += 1;
                 block_indices_begins_data += 1;
             }
+            sequence_group->set_seq_len_to_sample(matmul_gathering_is_required ? std::min(actual_seq_len, num_scheduled_tokens) : num_scheduled_tokens);
```

Comment:
```suggestion
            sequence_group->set_seq_len_to_sample(seq_len);
```
why not just as simple as suggested?